File: gridfly-quickstart-dot-net/SyncApiClient.cs
using System.IO;
using System.Threading.Tasks;
using Gridfly.Client;

namespace Gridfly.QuickStart
{
    public class SyncApiClient : BaseApiClient
    {
        public SyncApiClient(string clientId, string clientSecret) : base(clientId, clientSecret)
        {
        }

        public async Task<byte[]> Export(Stream htmlStream)
        {
            var client = await Authenticate();
            try
            {
                var response = await client.CreateSyncJobAsync(htmlStream);
                using (var ms = new MemoryStream())
                {
                    await response.Stream.CopyToAsync(ms);
                    return ms.ToArray();
                }
            }
            catch (ApiException ex)
            {
                Console.WriteLine($"Sync Export Failed. Status: {ex.StatusCode}");
                Console.WriteLine($"Response: {ex.Response}");
                throw;
            }
        }
    }
}

File: gridfly-quickstart-java/src/main/java/io/gridfly/quickstart/client/SyncApiClient.java
package io.gridfly.quickstart.client;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.gridfly.client.generated.ApiClient;
import io.gridfly.client.generated.ApiException;
import io.gridfly.client.generated.api.SyncJobCreationApi;
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.StandardCopyOption;

public class SyncApiClient extends BaseApiClient {

  public void generateExcel(
      String clientId, String clientSecret, File sourceFile, File outputDirectory)
      throws IOException {

    try {
      ApiClient client = authenticate(clientId, clientSecret);
      SyncJobCreationApi syncJobCreationApi = new SyncJobCreationApi(client);
      File generated = syncJobCreationApi.createSyncJob(sourceFile);
      File permanentFile = new File(outputDirectory, generated.getName());
      Files.move(generated.toPath(), permanentFile.toPath(), StandardCopyOption.REPLACE_EXISTING);

      System.out.println("Export completed: file saved to: " + permanentFile.getAbsolutePath());

    } catch (ApiException ex) {
      ObjectMapper mapper = new ObjectMapper();
      String responseBody = ex.getResponseBody();
      String formattedResponseBody =
          responseBody == null || responseBody.isEmpty()
              ? ""
              : (mapper
                  .writerWithDefaultPrettyPrinter()
                  .writeValueAsString(mapper.readValue(ex.getResponseBody(), Object.class)));

      System.out.println(
          "API returned error with status code " + ex.getCode() + ":\n" + formattedResponseBody);
    }
  }
}
